sysl2/sysl/fmt_grammar/FmtParser.g4

```diff
@@ -0,0 +1,60 @@
+grammar FmtParser;
+
+expression
+    : text* ((simp_exp
+                 | bool_exp
+                 | cmp_exp
+                 | text) text*)*
+    ;
+yes_stmt
+    : text* ((simp_exp
+                 | bool_exp
+                 | cmp_exp
+                 | text) text*)*
+    ;
+no_stmt
+    : text* ((simp_exp
+                 | bool_exp
+                 | cmp_exp
+                 | text) text*)*
+    ;
+cmp_exp_body
+    : param eq value QM yes_stmt (BAR no_stmt)?
+    ;
+cmp_exp
+    : EXP_OPEN cmp_exp_body EXP_CLOSE
+    ;
+pattern_exp_body
+    :param PATTERN value QM yes_stmt (BAR no_stmt)?
+    ;
+bool_exp_body
+    :param QM yes_stmt (BAR no_stmt)?
+    ;
+bool_exp
+    : EXP_OPEN (bool_exp_body|pattern_exp_body) EXP_CLOSE
+    ;
+simp_exp
+    : EXP_OPEN param EXP_CLOSE
+    ;
+
+text
+    : TSTR | STRING | QSTRING
+    ;
+param
+    : TSTR | STRING | QSTRING
+    ;
+value:QSTRING;
+eq   :DEQ;
+
+EXP_OPEN    : '%(' ;
+EXP_CLOSE: ')';
+QM : '?';
+DEQ: '==' | '!=';
+PATTERN:'~';
+BAR: '|';
+AT: '@';
+TSTR:('<color red>' | '<color green>') (~[%<]+ | '%' ~[(]+) | '</color>' | STRING TSTR;
```

Comment:
why is color hardcoded here?

Reply:
For complex case like `%(@ggg?//«%(@ggg)»//**%(patterns? %(pat~/\btba|tbd\b/?<color red>%(pat)</color>|<color green>%(pat)</color>) | <color red>pat?</color>)**\n|%(needs_int?<color red>(missing INT%)</color>\n))`
HTML tag may contain some keywords that I need to escape. I need to know its a HTML tag, but 
do I need to write whole grammar for HTML? Or just some special cases we need to use.

Reply:
Or we can use `\` to escape all the keywords, in that way, will change the way we write the appfmt and epfmt.

Reply:
@JackLeeHal what I mean is why hardcode `red` or `green`? what if i want `blue` or `yellow`?

Reply:
Also i dont think you should have `<color>..</color> as part of fmt parser grammar. it is just text for plantuml.